diff --git a/distrib/sets/checkflist b/distrib/sets/checkflist
index b3b1737..43ee3df 100644
--- a/distrib/sets/checkflist
+++ b/distrib/sets/checkflist
@@ -29,12 +29,15 @@ if [ "X${DESTDIR}" = "X" ]; then
 	exit 1
 fi
 
-Release=$1
-
+diffarg=""
+if [ "$1" = "-u" ]; then
+	diffarg="-u"
+	shift
+fi
 
-TMP=`mktemp -d /tmp/_xckr.XXXXXXXXXX` || exit 1
-trap 'rm -rf $TMP' 0 1 15
+TMP=`mktemp /tmp/_xckr.XXXXXXXXXX` || exit 1
+trap 'rm -f $TMP' 0 1 15
 
-sh makeflist $Release > $TMP/_CHECK
+sh makeflist > $TMP/_CHECK
 ( cd $DESTDIR ; find . \( -type d -o -type f -o -type l \) ) | \
-    grep -v -e '^\.$' | sort -H | diff -u $TMP/_CHECK -
+    grep -v -e '^\.$' | sort -H | diff $diffarg $TMP/_CHECK -
diff --git a/distrib/sets/gensetlist b/distrib/sets/gensetlist
index 418ffc5..4f7c6d7 100644
--- a/distrib/sets/gensetlist
+++ b/distrib/sets/gensetlist
@@ -5,7 +5,6 @@ arch=$MACHINE
 listsdir=`pwd`/lists
 
 setname=$1
-Release=$2
 
 if [ "$setname" = "xfont" ]; then
         arch=x11r7
diff --git a/distrib/sets/makeflist b/distrib/sets/makeflist
index b1be993..d08fb55 100644
--- a/distrib/sets/makeflist
+++ b/distrib/sets/makeflist
@@ -2,11 +2,10 @@
 #	$OpenBSD: makeflist,v 1.1.1.1 2006/11/27 11:25:45 matthieu Exp $
 
 arch=$MACHINE
-Release=$1
 listsdir=`pwd`/lists
 
 lists=`sh ./genlists $arch`
 for setname in $lists
 do
-	sh ./gensetlist $setname $Release
+	sh ./gensetlist $setname
 done | sort
diff --git a/distrib/sets/maketars b/distrib/sets/maketars
index 7340287..b0b5ae1 100644
--- a/distrib/sets/maketars
+++ b/distrib/sets/maketars
@@ -3,8 +3,6 @@
 
 # xy
 RELEASE=$1
-Release=$2
-# x.y
 
 if [ "X${RELEASE}" = "X" ]; then
 	echo "Usage: `basename $0` <OSREVISION>"
@@ -37,7 +35,7 @@ cd $DESTDIR
 
 for setname in $lists; do
 	echo -n "${setname}: "
-	(cd $pwd; sh gensetlist $setname $Release) | sort > $TMP1
+	(cd $pwd; sh gensetlist $setname) | sort > $TMP1
 	sed -n  '/^\.\/usr\/X11R6\/lib\/lib.*\.so\..*/{p;}' $TMP1 > $TMP2
 	sed     '/^\.\/usr\/X11R6\/lib\/lib.*\.so\..*/d'    $TMP1 >> $TMP2
 	pax -w -d < $TMP2 | gzip -9 > ${tardir}/${setname}${RELEASE}.tgz
